source/_docs/blueprint/tutorial.markdown

```diff
@@ -0,0 +1,229 @@
+
+The first step is to add a name and description to the fields to help identify the fields.
+
+```yaml
+blueprint:
+  name: Motion Light Tutorial
+  domain: automation
+  input:
+    motion_sensor:
+      name: Motion Sensor
+      description: This sensor will be synchronized with the light.
+    target_light:
+      name: Lights
+      description: The lights to keep in sync.
+```
+
+## Describing the configurable parts
+
+Our blueprint doesn't currently describe what the configurable part should contain. Without this information Home Assistant will offer the user an empty text box.
+
+To do this, we will use selectors. Selectors describe a type and can be used to help the user pick a value.
+
+The selector for the motion sensor entity should describe that we want entities from the binary sensor domain that have the device class `motion`.
+
+The selector for the target light should describe that we want to target light entities.
+
+```yaml
+blueprint:
+  name: Motion Light Tutorial
+  domain: automation
+  input:
+    motion_sensor:
+      name: Motion Sensor
+      description: This sensor will be synchronized with the light.
+      selector:
+        entity:
+          domain: binary_sensor
+          device_class: motion
+    target_light:
+      name: Lights
+      description: The lights to keep in sync.
+      selector:
+        target:
+          entity:
+            domain: light
+```
+
+## The final blueprint
+
+After we have added all the steps, our blueprint will look like this:
+
+{% raw %}
+```yaml
+blueprint:
+  name: Motion Light Tutorial
+  domain: automation
+  input:
+    motion_sensor:
+      name: Motion Sensor
+      description: This sensor will be synchronized with the light.
+      selector:
+        entity:
+          domain: binary_sensor
+          device_class: motion
+    target_light:
+      name: Lights
+      description: The lights to keep in sync.
+      selector:
+        target:
+          entity:
+            domain: light
+
+trigger:
+  platform: state
+  entity_id: !input motion_sensor
+
+action:
+  service: >
+    {% if trigger.to_state.state == "on" %}
+      light.turn_on
+    {% else %}
+      light.turn_off
+    {% endif %}
+  target: !input target_light
+```
+{% endraw %}
+
+## Use it via the UI
+
+To configure it via the UI, go to **Configuration** and then **Blueprints**. Find the "Motion Light Tutorial" blueprint and click on "Create Automation".
+
+<p class='note'>
+Don't forget to reload automations after you make changes to your blueprint to have the UI and the automation intgration pick up the latest blueprint changes.
+</p>
+
+![Screenshot of the blueprint  UI](/images/blueprints/tutorial-ui.png)
+
+## Share the love
+
+The final step is to share this blueprint with others. For this tutorial we're going to share it on GitHub Gists.
+
+- Go to [GitHub Gists](https://gist.github.com/)
+- Gist description: blueprint tutorial
+- Filename including extension: `motion_light_tutorial.yaml`
```

Comment:
```suggestion
- File name including extension: `motion_light_tutorial.yaml`
```

Reply:
This is a direct copy of the text they use:

<img width="293" alt="image" src="https://user-images.githubusercontent.com/1444314/101139689-33205400-3612-11eb-87d4-458e4a0e93da.png">

Reply:
Then it should stay the way the write it (although that it rather uncommon spelling according to Google). But apparently even Oxford does allow it, same with "username".

Reply:
Yeah, that is a GitHub typo.

- <https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/f/file-name>
- <https://developers.google.com/style/word-list#letter-f>